sodiumpp/include/sodiumpp/sodiumpp.h

```diff
@@ -338,7 +338,7 @@ inline void check_valid_size(T1 current, T2 expected, const char * name, const c
         /**
          * Construct a secret key from a pregenerated public and secret key.
          */
-        secret_key(const public_key<P>& pk, const encoded_bytes& secret_bytes) : pk(pk), secret_bytes(locked_string::move_from_not_locked_string(secret_bytes.to_binary())) {}
+        secret_key(const public_key<P>& pk, const encoded_bytes& secret_bytes) : pk(pk), secret_bytes(secret_bytes.to_binary()) {}
```

Comment:
We need to change API of this, creating from encoded bytes of secret key already is a mistake and does not make sense,
already the memory leaked at this point.

API of secret key should be to just be created only from already locked string.
(and provide named unsafe constructor for people who wish to use this but do not wish to use locked string)